Assets/UI/WorldView/citybannermanager_CQUI.lua

```diff
@@ -239,9 +217,65 @@ end
 -- ============================================================================
 -- CQUI Replacement Functions (Common to both basegame and expansions)
 -- ============================================================================
+function OnCityBannerClick( playerID, cityID )
+    print_debug("CityBannerManager_CQUI: OnCityBannerClick ENTRY  playerID:"..tostring(playerID).." cityID:"..tostring(cityID));
+    local pPlayer = Players[playerID];
+    if (pPlayer == nil) then
+        return;
+    end
+
+    local pCity = pPlayer:GetCities():FindID(cityID);
+    if (pCity == nil) then
+        return;
+    end
+
+    if (g_bIsRiseAndFall or g_bIsGatheringStorm) then
+        if (pPlayer:IsFreeCities()) then
+            UI.LookAtPlotScreenPosition( pCity:GetX(), pCity:GetY(), 0.5, 0.5 );
+            return;
+        end
+    end
+    
+    local localPlayerID;
+    if (WorldBuilder.IsActive()) then
+        localPlayerID = playerID;   -- If WorldBuilder is active, allow the user to select the city
+    else
+        localPlayerID = Game.GetLocalPlayer();
+    end
+
+    if (pPlayer:GetID() == localPlayerID) then
+        UI.SelectCity( pCity );
+        UI.SetCycleAdvanceTimer(0);     -- Cancel any auto-advance timer
+        UI.SetInterfaceMode(InterfaceModeTypes.CITY_MANAGEMENT);
+    elseif (localPlayerID == PlayerTypes.OBSERVER 
+            or localPlayerID == PlayerTypes.NONE 
+            or pPlayer:GetDiplomacy():HasMet(localPlayerID)) then
+        LuaEvents.CQUI_CityviewDisable(); -- Make sure the cityview is disable
+        local pPlayerConfig :table   = PlayerConfigurations[playerID];
```

Comment:
Why is this entire section about Minor Civ here?

Reply:
except for line 253 (LuaEvents.CQUI_CityviewDisable();) it's the same code from the unmodified OnCityBannerClick in CityBannerManager.lua (both expansions and vanilla)

ditto about testing, I've played maybe 2 MP Civ games in my life.  That player in the Steam Workshop thread set up a MP game and I played for a bit, unable to repro the issue.  Apparently it just "eventually breaks" and I couldn't get it to eventually break, but I also didn't have a ton of time to test.  I relied upon them trying this out and he said they found no issues as of yet.